src/odemis/gui/cont/fastem_acq.py

```diff
@@ -431,6 +430,7 @@ def _reset_acquisition_gui(self, text=None, level=None):
         else:
             self.update_acquisition_time()
 
+    @call_in_wx_main  # call in main thread as changes in GUI are triggered
```

Comment:
No, `on_acquisition()` is called when the "acquire" button is pressed. (see the "evt" argument). So it's already called in the main GUI thread. It shouldn't be called delayed.